### PR TITLE
Added conforming datatypes & copatibility info

### PR DESCRIPTION
**Pull request by BBrandl948**
### Description

added som Information about related protocols and compatible data types

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [ X] All writings are my own.
- [ X] My entry follows the Codecademy Docs style guide.
- [ X] My changes generate no new warnings.
- [ X] I have performed a self-review of my own writing and code.
- [ X] I have checked my entry and corrected any misspellings.
- [ X] I have made corresponding changes to the documentation if needed.
- [ X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:2]: The changes to the metadata break the entry and should be removed. You should review the rules on metadata [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#metadata) before attempting to edit another entry's metadata. Please revert these changes to the metadata.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:17]:
```suggestion
**`Encodable`** is a protocol that enables a type's data to be encoded for use with Application Programming Interfaces ([APIs](https://www.codecademy.com/resources/docs/general/api)).
```
I would like to revert this change.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:24]:
```suggestion
> **Note:** This protocol is related to the following protocols:
> - Codeable
> - Decodeable
> - CodingKey
> - CodingKeyRepresentable
> - CodingUserInfoKey
```

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:17]: I'd like you not to delete this line.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:37]: This entire section is a large infodump in the middle of the entry. Docs entries are supposed to be relatively short and easy to digest. I don't think this list is adding anything to the entry. I'd remove it.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@BBrandl948 I did a review and left you some comments.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (BBrandl948) [reply content/swift/concepts/protocols/terms/encodable/encodable.md:37]: I deleted all lines. I think, that I had too little knowledge about Codecademy docs and its intentions when I started to work on this article. So I simply added Information, that seemed to be important to me. It was a demanded exercise in the iOS development career path and I could not find out in the description what exactly should be covered in the articles. So I simply jumped in.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (BBrandl948) [reply content/swift/concepts/protocols/terms/encodable/encodable.md:17]: I added the line again in the document.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (BBrandl948) [reply content/swift/concepts/protocols/terms/encodable/encodable.md:2]: I restored the whole section. Sorry for distracting the intended structure. I was focused on the layout and unfortunately did not realize breaking the entry's metadata.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (BBrandl948) [reply content/swift/concepts/protocols/terms/encodable/encodable.md:37]: I made all changes in my local repo. In case that nearly all of my adds do not really match, I would suggest to simply delete my branch. If you do not agree with me please let me know what to do next. Thank you for your understanding.

**Comment by SSwiniarski** [comment]
@BBrandl948 I don't see any of your changes. Have you pushed your changes to your remote branch?

**Comment by BBrandl948** [comment]
First I made the changes only in my local repo. Now you should be able to see them here too

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:7]:
```suggestion
  - 'Mobile Development'
```

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:2]:
```suggestion
Title: 'Encodable'
```

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:18]:
```suggestion
**`Encodable`** is a protocol that enables a type's data to be encoded and for use with Application Programming Interfaces ([APIs](https://www.codecademy.com/resources/docs/general/api)). If data also needs to be decoded, it's convenient to use the `Codable` protocol because it includes both the `Encodable` and `Decodable` protocols.

The protocol is related to the following protocols:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@BBrandl948 I left a few more comments.

**Comment by BBrandl948** [comment]
Totally agree with all of your suggestions and committed all of them

**Review by SSwiniarski — APPROVED** [review]
@BBrandl948 Looks good for a second review.

**Comment by BBrandl948** [comment]
Thank you for your great help.

**Comment by BBrandl948** [comment]
Let me know when I can close the pull request.

**Comment by SSwiniarski** [comment]
@BBrandl948 you shouldn't close the pull request unless you're withdrawing it from consideration.

**Review by yangc95 — CHANGES_REQUESTED** [review]
@BBrandl948 left one comment for the second review! Please tag me when finished or if you need further assistance 🙂

**Comment by BBrandl948** [comment]
I committed all the requested changes and I'm waiting for your feedback

**Comment by BBrandl948** [comment]
@yangc95 forgot to tag you. I committed all changes.

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (yangc95) [inline content/swift/concepts/protocols/terms/encodable/encodable.md:22]: @BBrandl948 Can you include a brief sentence for each about how it is related

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (yangc95) [reply content/swift/concepts/protocols/terms/encodable/encodable.md:17]: @BBrandl948 It's _best left as a note_ to differentiate its content from the rest of the introduction, since it's something the reader should be aware of outside the term

**Comment by yangc95** [comment]
Thanks for that @BBrandl948! I have a few more comments

File: content/swift/concepts/protocols/terms/encodable/encodable.md
Comment (BBrandl948) [reply content/swift/concepts/protocols/terms/encodable/encodable.md:22]: I will try to

Comment (BBrandl948) [reply content/swift/concepts/protocols/terms/encodable/encodable.md]: I hope my suggestions meet your intentions. Committed all changes

**Comment by BBrandl948** [comment]
@yangc95  I committed all changes I made in the document.

**Review by yangc95 — APPROVED** [review]

**Pull request merged**
